engine/class_modules/priest/sc_priest.cpp

```diff
@@ -2157,10 +2209,19 @@ void priest_t::init_rng()
 
 void priest_t::init_background_actions()
 {
-  background_actions.echoing_void = new actions::spells::echoing_void_t( *this );
-
-  init_background_actions_shadow();
-  init_background_actions_discipline();
+  if ( specialization() == PRIEST_SHADOW )
+  {
+    background_actions.echoing_void = new actions::spells::echoing_void_t( *this );
+    init_background_actions_shadow();
+  }
+  else if ( specialization() == PRIEST_DISCIPLINE )
```

Comment:
I am not the biggest fan of having to use specialization() checks everywhere in initialization. If you correctly conditionalize things on spell data being enabled (see my comments on background actions holy fire and searing light), you don't need artificial specialization() checks. This makes things also just work out if you have a background action that is available for both discipline and holy, let's say based on a shared talent, without having yo create yet another condition branch.

Same applies for buff and debuff effects.

That said, I know that they can make life easier and a lot of other class modules do it that way just as well. I don't really care that much, but thought it worth mentioning.

@seanpeters86 what is your opinion on this?

Reply:
I generally agree, I think the check here probably isnt needed @taherbert ?

Reply:
Agree, check here not needed. I think we may still need the check around `parse_buff_effects` to de-clutter the report, though.